### PR TITLE
Setup `mdbook` checks and GitHub pages

### LINKED ISSUE #13: Add CI for mdbook
Just make sure everything builds correctly for now.

### LINKED ISSUE #14: Host the mdbook on GH Pages
This would be nice for casual observers and artists / designers to quickly read what we've written. Doesn't need to be super robust or pretty.

### PR DESCRIPTION
**Pull request by BD103**
Closes #13 and closes #14.

This PR does two things:

1. Runs `mdbook build` and `mdbook test` on all commits, PRs, and anything else that triggers `cy.yaml`.
    - The latter, `mdbook test`, checks that all Rust code snippets compile! Neat!
2. Runs `mdbook build` and deploys the results to GitHub Pages whenever a new commit is pushed to `main`.

### CODE REVIEW

**Comment by BD103** [comment]
Before this can be merged, a maintainer will have to go to <https://github.com/bevyengine/bevy_editor_prototypes/settings/pages> and set the "Source" dropdown to "GitHub Actions". :)

**Review by alice-i-cecile — APPROVED** [review]
You're the best.

**Pull request merged**
